lib/internal/url.js

```diff
@@ -767,7 +766,7 @@ ObjectDefineProperties(URLSearchParams.prototype, {
  * @returns {self is URL}
  */
 function isURL(self) {
-  return Boolean(self?.href && self.protocol && self.auth === undefined && self.path === undefined);
+  return typeof self === 'object' && self.href && self.protocol && self.auth === undefined && self.path === undefined;
```

Comment:
Unfortunately this introduces a regression on `isURL(null)` ... with the previous implementation `isURL(null)` returns false, with the new impl it throws. You would need to include an additional `&& self !== null` before the `self.href` check.

Reply:
HA! jinx...

Reply:
I addressed these issues in my latest commit.

Reply:
> HA! jinx... 

@jasnell

![image](https://github.com/nodejs/node/assets/1935246/29530769-7cea-4df0-9b98-7ec85e2f011f)